src/hiccup/compiler.clj

```diff
@@ -49,10 +66,17 @@
   (or content
       (and (html-mode?) (not (void-tags tag)))))
 
+(defn class-list [value]
```

Comment:
That makes sense, but I think the way you go about it could be a little more performant. Perhaps something like:

```clojure
(defn- merge-classes [class classes]
  (cond
    (nil? class)    classes
    (string? class) (str class " " classes)
    :else           (str (str/join " " class) " " classes)

(defn- merge-attributes [map-attrs id classes]
  (-> map-attrs
      (cond-> id      (assoc :id id))
      (cond-> classes (assoc :class (merge-classes (:class map-attrs) classes)))))

 (defn normalize-element
   "Ensure an element vector is of the form [tag-name attrs content]."
   [[tag & content]]
   (when-not (or (keyword? tag) (symbol? tag) (string? tag))
      (throw (IllegalArgumentException. (str tag " is not a valid element name."))))
    (let [[_ tag id class] (re-matches re-tag (util/as-str tag))
          classes          (if class (str/replace class "." " "))
          map-attrs        (first content)]
      (if (map? map-attrs)
        [tag (merge-attributes map-attrs id classes) (next content)]
        [tag {:id id, :class classes} content])))